src/sign_dlink_ru.c

```diff
@@ -183,15 +187,27 @@ FILE* make_out_file(char* filename) {
  * Sign the firmware file after all of our checks have completed
  */
 void sign_firmware(char* filename, char** salts, int num_salts) {
-	int i;
+	int i = 0;
 	salt_t salt;
 	FILE* f = make_out_file(filename);
+	
+	char* version_suffix = "c0ffeef0rge";
+			
 
+	if (strcmp(salts[0],"-v") == 0) {
```

Comment:
If an arg starting with `-` is specified but not `-v`, it will treat it as a hex (even if not hex)

Reply:
as -v was the only option it seemed ok to error on anything else but can change

and atm is in a fixed position witch I don't like